.github/workflows/rocks.yml

```diff
@@ -0,0 +1,60 @@
+name: Rocks server
+on:
+  schedule:
+    - cron: '0 */3 * * *'
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - '.github/workflows/rocks.yml'
+
+# Cancel workflow runs on PRs when the PR is updated with a newer commit.
+# Such runs will have a concurrency group named
+# `{github.workflow}-{github.ref}`,
+# for example,
+# `Amazon Linux-refs/pull/42/merge`.
+#
+# Runs on branch `master` and tags will never be canceled,
+# due to having a unique group name
+# `{github.run_id}-{github.run_attempt}`,
+# for example,
+# `3477882280-1`.
+concurrency:
+  group: ${{
+    (
+    github.ref == 'refs/heads/master' ||
+    startsWith(github.ref, 'refs/tags/')
+    ) &&
+    format('{0}-{1}', github.run_id, github.run_attempt) ||
+    format('{0}-{1}', github.workflow, github.ref) }}
+  cancel-in-progress: true
+
+jobs:
+  check-rocks-server:
+    runs-on: [ self-hosted, ubuntu-20.04-self-hosted ]
```

Comment:
I think there is no need to run this job on a self-hosted runner. You can use `ubuntu-22.04`.

Reply:
Done